### ISSUE TITLE
Compilation error when PostgreSQL stored function has an ANY[] data type reference

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13163]
A PostgreSQL function can work like this:

```sql
CREATE FUNCTION f_anyarray(in_array IN anyarray)
RETURNS anyarray ...
```

We don't really support those yet (see https://github.com/jOOQ/jOOQ/issues/5479), but due to a recent fix in the code generator, there's now an illegal data type reference `SQLDataType.null` in generated output. Seems this regression hasn't gone live yet.

**Comment by lukaseder** [comment]
Yeah, the regression happened because of these array type fixes: https://github.com/jOOQ/jOOQ/issues/13107, https://github.com/jOOQ/jOOQ/issues/13073, https://github.com/jOOQ/jOOQ/issues/13097

**Comment by lukaseder** [comment]
I think this regression is specific to the `ANYARRAY` type, which jOOQ does't recognise as `ANY[]`. Seems like a weird part of the PG type system...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13164), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13165)
